File: Front-end/src/app/events/meta-models/associated-event.ts
import { Lookup } from '../../lookups/lookup';
import { Severity } from '../../shared/enums/severity';
import { Status } from '../../shared/enums/status';
export interface AssociatedEvent {
		category: Lookup;
		type: Lookup;
		severity: Severity;
		effectRange: {
			affectedRadius?: number;
			nearRadius?: number;
			location: Location;
        };
        categoryStatus: Status;
}

File: Service/Z2data.Core/Domain/Events/AssociatedEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Z2data.Core.Domain.Lookups;
using Z2data.Core.Enums;

namespace Z2data.Core.Domain.Events
{
    public class AssociatedEvent
    {
        public Lookup Category { get; set; }
        public Lookup Type { get; set; }
        public Severity Severity { get; set; }
        public EffectRange EffectRange { get; set; }
        public Status CategoryStatus { get; set; }
    }
}
